### PR TITLE
Subclass InputBarView to allow picking images

### LINKED ISSUE #1511: Allow to pick an image using UIImagePickerController
Hi everyone
It's really an awesome repo which helps me a lot  and I would like to contribute to,  what I need to do adding a camera button into documentation example to allow pick an Image using  UIImagePickerController as shown in the screenshot below

CameraButton

I've implemented it so can I make a pull request?
thank you in advance

### PR DESCRIPTION
**Pull request by MohannadBakbouk**
What does this implement/fix? Explain your changes.
---------------------------------------------------

Two weeks ago I've used this great library in my app so I wanted to allow user for picking up an image, I did some research about how to integrate it with MessageKit  but  unfortunately, I didn't  find any good solutions, 
I decied to contribute to this awesome library, so  I'm going to illustrate how we can implement adding photo feature in this documentation example.
My Initial solution was just corporate UIImagePickerController to ChatviewController and Implement the delegation there
but @Kaspik  point me to a better way in issue 1511, as a result, I've subclasses InputBarAccessoryView and Implemented the following items
1. AttchmentsManager plugin 
2. UIImagePickerControllerDelegate
3. subclass InputBarAccessoryViewDelegate to allow receiving attachments array at didPressSendButtonWith attachments
finally, I've used my approach in AdvancedController
Simulator Screen Shot - X - 2020-12-25 at 09 59 50
Simulator Screen Shot - X - 2020-12-25 at 10 00 04
Simulator Screen Shot - X - 2020-12-25 at 10 00 15

Does this close any currently open issues?
------------------------------------------
Closes #1511 

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Devices/Simulators:** …Simulators

**iOS Version:**14.2

**Swift Version:**5.3

**MessageKit Version:**3.4.2

### CODE REVIEW

File: Example/Sources/Extensions/AlertService.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+class AlertService {
+
+    
+    static func showAlert(style: UIAlertController.Style, title: String?, message: String?, actions: [UIAlertAction] = [UIAlertAction(title: "Ok", style: .cancel, handler: nil)], completion: (() -> Swift.Void)? = nil) {
+        let alert = UIAlertController(title: title, message: message, preferredStyle: style)
+        for action in actions {
+            alert.addAction(action)
+        }
+        if let topVC = UIApplication.getTopMostViewController() {
```
Comment (Kaspik) [inline Example/Sources/Extensions/AlertService.swift]: If this doesn§t exist, which means sourceView is nil, doesn't this crash on iPad?

File: Example/Sources/Extensions/AlertService.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            alert.popoverPresentationController?.sourceView = topVC.view
+            alert.popoverPresentationController?.sourceRect = CGRect(x: topVC.view.bounds.midX, y: topVC.view.bounds.midY, width: 0, height: 0)
+            alert.popoverPresentationController?.permittedArrowDirections = []
+            topVC.present(alert, animated: true, completion: completion)
```
Comment (Kaspik) [inline Example/Sources/Extensions/AlertService.swift]: Also, if it's nil, it's not presented at all?

File: Example/Sources/Resources/Assets.xcassets/camera.imageset/Contents.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
```
Comment (Kaspik) [inline Example/Sources/Resources/Assets.xcassets/camera.imageset/Contents.json]: Can we either use PDF file in vectors, or use SF symbols (for iOS 13+) and system icon (https://developer.apple.com/documentation/uikit/uiapplicationshortcuticon/icontype/capturephoto for iOS 12)? (or https://developer.apple.com/documentation/uikit/uibarbuttonitem/systemitem/camera) - check https://developer.apple.com/design/human-interface-guidelines/ios/icons-and-images/system-icons/

**Review by Kaspik — COMMENTED** [review]
Just a nitpicks, nice, thanks for these examples! :)

File: Example/Sources/Extensions/AlertService.swift (hunk 1 above)
Comment (MohannadBakbouk) [reply Example/Sources/Extensions/AlertService.swift]: I did an optional binding to prevent crashing,  and I've just tested it on iPad it worked

Comment (MohannadBakbouk) [reply Example/Sources/Extensions/AlertService.swift]: do you want me to replace  UIApplication.getTopMostViewController()  with 
"(UIApplication.shared.delegate as? AppDelegate)?.window?.rootViewController?." ?? but It's optional value too and it needs to check

Comment (Kaspik) [reply Example/Sources/Extensions/AlertService.swift]: I'm happy it works, I'm just thinking about edge case where `topVC` is nil, which means nothing happens and alert is not presented, right?

**Review by Shanebitch — APPROVED** [review]

**Comment by mlyongardiner** [comment]
This looks really cool and extremely helpful for something I'm working on as well. Would you say this is production-ready? Thanks!

**Comment by MohannadBakbouk** [comment]
@mlyongardiner yes it's ready, I wanna fix an issue and send it

**Comment by mlyongardiner** [comment]
@MohannadBakbouk please let me know when you send it!!

File: Example/Sources/Resources/Assets.xcassets/camera.imageset/Contents.json (hunk 3 above)
Comment (MohannadBakbouk) [reply Example/Sources/Resources/Assets.xcassets/camera.imageset/Contents.json]: @Kaspik 
I've changed to SF Symbols so I made a condition to check if the app run on ios 13 then use System-icon

File: Example/Sources/Extensions/AlertService.swift (hunk 2 above)
Comment (MohannadBakbouk) [reply Example/Sources/Extensions/AlertService.swift]: I replaced the whole block with a new way 
I've just  pushed my modifications

**Comment by MohannadBakbouk** [comment]
@Kaspik 
I've modified them as you had mentioned  so please take a look

File: Example/Sources/Extensions/AlertService.swift [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+class AlertService {
+
+    
+    static func showAlert(style: UIAlertController.Style, title: String?, message: String?, actions: [UIAlertAction] = [UIAlertAction(title: "Ok", style: .cancel, handler: nil)], completion: (() -> Swift.Void)? = nil) {
+        let alert = UIAlertController(title: title, message: message, preferredStyle: style)
+        for action in actions {
+            alert.addAction(action)
+        }
+//        if let topVC = UIApplication.getTopMostViewController() {
```
Comment (Kaspik) [inline Example/Sources/Extensions/AlertService.swift]: Let's remove this

File: Example/Sources/Extensions/AlertService.swift [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+//            alert.popoverPresentationController?.sourceView = topVC.view
+//            alert.popoverPresentationController?.sourceRect = CGRect(x: topVC.view.bounds.midX, y: topVC.view.bounds.midY, width: 0, height: 0)
+//            alert.popoverPresentationController?.permittedArrowDirections = []
+//            topVC.present(alert, animated: true, completion: completion)
+//        }
+        
+        if let current = (UIApplication.shared.delegate as? AppDelegate)?.window?.rootViewController {
```
Comment (Kaspik) [inline Example/Sources/Extensions/AlertService.swift]: Let's simply do `UIApplication.shared.delegate?.window??.rootViewController?.present(alert, animated: true, completion: completion)`? There is no need for the `AppDelegate` `as?`?

File: Example/Sources/Extensions/AlertService.swift [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        }
+    }
+    
+    
+}
+
+
+
+extension UIApplication {
+    class func getTopMostViewController(base: UIViewController? = UIApplication.shared.keyWindow?.rootViewController) -> UIViewController? {
```
Comment (Kaspik) [inline Example/Sources/Extensions/AlertService.swift]: can be removed?

File: Example/Sources/Resources/Assets.xcassets/camera.imageset/Contents.json (hunk 3 above)
Comment (Kaspik) [reply Example/Sources/Resources/Assets.xcassets/camera.imageset/Contents.json]: Can we use `UIApplicationShortcutIconTypeCapturePhoto` for iOS 12 and remove this asset completely?

**Review by Kaspik — CHANGES_REQUESTED** [review]
Final polish! Thanks :)

**Comment by mlyongardiner** [comment]
@MohannadBakbouk Excited for this to be merged!! @Kaspik thanks for your help here this will be an awesome feature addition.

**Comment by MohannadBakbouk** [comment]
@Kaspik 
thank you for your supporting
I've changed the points you had  mentioned

**Review by Kaspik — APPROVED** [review]

**Pull request merged**